homeassistant/components/fritzbox/light.py

```diff
@@ -129,16 +132,33 @@ async def async_turn_on(self, **kwargs: Any) -> None:
             level = kwargs[ATTR_BRIGHTNESS]
             await self.hass.async_add_executor_job(self.device.set_level, level)
         if kwargs.get(ATTR_HS_COLOR) is not None:
-            hass_hue = int(kwargs[ATTR_HS_COLOR][0])
-            hass_saturation = round(kwargs[ATTR_HS_COLOR][1] * 255.0 / 100.0)
-            # find supported hs values closest to what user selected
-            hue = min(self._supported_hs.keys(), key=lambda x: abs(x - hass_hue))
-            saturation = min(
-                self._supported_hs[hue], key=lambda x: abs(x - hass_saturation)
-            )
-            await self.hass.async_add_executor_job(
-                self.device.set_color, (hue, saturation)
-            )
+            # Try setunmappedcolor first. This allows free color selection,
+            # but we don't know if its supported by all devices.
+            try:
+                # HA gives 0..360 for hue, fritz light only supports 0..359
+                unmapped_hue = int(kwargs[ATTR_HS_COLOR][0] % 360)
+                unmapped_saturation = round(kwargs[ATTR_HS_COLOR][1] * 255.0 / 100.0)
+                await self.hass.async_add_executor_job(
+                    self.device.set_unmapped_color, (unmapped_hue, unmapped_saturation)
+                )
+            # This will raise 400 BAD REQUEST if the setunmappedcolor is not available
+            except HTTPError as err:
+                if err.response.status_code != 400:
+                    raise
+                LOGGER.debug(
+                    "fritzbox does not support method 'setunmappedcolor', fallback to 'setcolor'"
+                )
```

Comment:
If you're not sure if this is supported by all hardware or not, you could consider making this an info or warning requesting users to open an issue specifying the model of the light they're using. Make sure that's only logged once per light though.

This is how cast asks users to open an issue about unknown chrome cast models:
https://github.com/home-assistant/core/blob/1e4690626ff7e0e978805d9fa7b4dc7270811711/homeassistant/components/cast/helpers.py#L59-L96

This can be an improvement for a follow-up PR though.